core/git-access.c

```diff
@@ -692,15 +691,11 @@ static git_repository *create_and_push_remote(const char *localdir, const char *
 
 	/* create a config so we can set the remote tracking branch */
 	git_repository_config(&conf, repo);
-	len = sizeof("branch..remote") + strlen(branch);
-	variable_name = malloc(len);
-	snprintf(variable_name, len, "branch.%s.remote", branch);
+	variable_name = format_string("branch.%s.remote", branch);
 	git_config_set_string(conf, variable_name, "origin");
 	/* we know this is shorter than the previous one, so we reuse the variable*/
-	snprintf(variable_name, len, "branch.%s.merge", branch);
-	len = sizeof("refs/heads/") + strlen(branch);
-	merge_head = malloc(len);
-	snprintf(merge_head, len, "refs/heads/%s", branch);
+	sprintf(variable_name, "branch.%s.merge", branch);
```

Comment:
i would slightly change the comment to be more verbose:
`/* "branch.%s.remote" is longer than "branch.%s.merge", so we can re-use the buffer. */`.

like you suggest, the clearer path for the reader would be to allocate a second buffer and free both at the end, but i don't think its that necessary. what we have is less readable, but functional as long as `strlen("remote") > strlen("merge")`.

Reply:
Good point - changed that.